packages/pages/src/upgrade/pagesUpdater.ts

```diff
@@ -43,46 +119,217 @@ const processDirectoryRecursively = (currentPath: string) => {
-    console.log(`Imports replaced in: ${filePath}`);
-  } catch (e) {
-    console.error("Error processing file: ", (e as Error).message);
+  };
+  processDirectoryRecursively(source, operation);
+};
+
+/**
+ * Replaces imports for pages/components with pages-components
+ * @param source the src folder
+ * @return hasReplaced
+ */
+export const replacePagesSlashComponentsImports = (source: string): boolean => {
+  let hasReplaced = false;
+  const operation = async (filePath: string) => {
+    const fileContent = fs.readFileSync(filePath, "utf8");
+    if (fileContent.match(pagesSlashComponentsRegex)) {
+      const modifiedContent = fileContent.replace(
+        pagesSlashComponentsRegex,
+        pagesComponentsReplacement
+      );
+      hasReplaced = true;
+      fs.writeFileSync(filePath, modifiedContent, "utf8");
+      console.log(`pages/components imports replaced in: ${filePath}`);
+    }
+  };
+  processDirectoryRecursively(source, operation);
+  return hasReplaced;
+};
+
+/**
+ * Replaces imports for sites-components with pages-components
+ * @param source the src folder
+ * @return hasReplaced
+ */
+export const replaceSitesComponentsImports = (source: string): boolean => {
+  let hasReplaced = false;
+  const operation = async (filePath: string) => {
+    const fileContent = fs.readFileSync(filePath, "utf8");
+    if (fileContent.match(sitesComponentsRegex)) {
+      const modifiedContent = fileContent.replace(
+        sitesComponentsRegex,
+        pagesComponentsReplacement
+      );
+      fs.writeFileSync(filePath, modifiedContent, "utf8");
+      hasReplaced = true;
+      console.log(`sites-components imports replaced in: ${filePath}`);
+    }
+  };
+  processDirectoryRecursively(source, operation);
+  return hasReplaced;
+};
+
+/**
+ * Replaces imports for react-components with pages-components
+ * @param source the src folder
+ * @return hasReplaced
+ */
+export const replaceReactComponentsImports = (source: string): boolean => {
+  let hasReplaced = false;
+  const operation = async (filePath: string) => {
+    const fileContent = fs.readFileSync(filePath, "utf8");
+    if (fileContent.match(reactComponentsRegex)) {
+      const modifiedContent = fileContent.replace(
+        reactComponentsRegex,
+        pagesComponentsReplacement
+      );
+      fs.writeFileSync(filePath, modifiedContent, "utf8");
+      hasReplaced = true;
+      console.log(`react-components imports replaced in: ${filePath}`);
+    }
+  };
+  processDirectoryRecursively(source, operation);
+  return hasReplaced;
+};
+
+/**
+ * Removes old fetch import that is no longer used
+ * @param root folder that contains package.json and tsconfig.json
+ */
+export const removeFetchImport = (root: string) => {
+  let hasRemoved = false;
+  const project = new Project({
+    tsConfigFilePath: path.resolve(root, "tsconfig.json"),
+  });
+  const sourceFiles = project.getSourceFiles();
+  for (let i = 0; i < sourceFiles.length; i++) {
```

Comment:
You need to check that fetch is coming from @yext/pages/util specifically. There could be other `fetch`es from other libs we don't want to remove.

Reply:
Add a test case for that too.